### PR TITLE
Add "back to home" CTA on 500 page (#1303)

### LINKED ISSUE #1303: [Bug] No "back to home" CTA on 500 page
## Description

https://app.appsmith.com/500 does not have any call to action for users to navigate back/reset their state.

## Solution

Add "back to home" button on 500 page. Design should be same as https://app.appsmith.com/404

### PR DESCRIPTION
**Pull request by JeffResc**
## Description
This PR adds a "Back to homepage" CTA on the [500 page](https://app.appsmith.com/500) to match the [400 page](https://app.appsmith.com/404).

Fixes #1303.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Using the internal tests available on GitHub.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by JeffResc** [comment]
The checklist mentions adding my own tests to test the functionality of this PR. Is something like that required for something minor like this?

File: app/client/src/pages/common/ServerUnavailable.tsx [hunk 1]
```diff
@@ -28,6 +28,16 @@ const ServerUnavailable = () => {
       <div>
         <p className="bold-text">Appsmith server is unavailable</p>
         <p>Please try again after some time</p>
+        <Button
```
Comment (hetunandu) [inline app/client/src/pages/common/ServerUnavailable.tsx:31]: Button is not defined

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/pages/common/ServerUnavailable.tsx (hunk 1 above)
Comment (JeffResc) [reply app/client/src/pages/common/ServerUnavailable.tsx:33]: Ah, got it, thanks 👍

**Comment by hetunandu** [comment]
Hey @JeffResc almost there, there are a few [prettier](https://prettier.io/) errors that is failing your build. Can we fix that? Looks like it is mainly indent issues
Screenshot 2020-10-23 at 2 11 52 PM

**Comment by JeffResc** [comment]
@hetunandu Thanks for letting me know, I looked through the PR checks and it's hard to find where you're getting that error from, I looked for a while. Anyway, I installed [Preitter for VSCode](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode), then configured and ran it, but it only seemed to change some formatting on line 12 rather than adjust the indentations. Is there a more preferred way to run Prettier to get it to output correctly?

**Comment by hetunandu** [comment]
Ideally this should be caught and fixed when you commit. Let me get back to you on this.

**Comment by hetunandu** [comment]
Ideally this should be caught and fixed when you commit. Let me get back to you on this.

**Comment by hetunandu** [comment]
@JeffResc Not being able to debug this as it seems to work properly on my side. Seems like we have some issue with our lint-staged plugin as this [file.](https://github.com/appsmithorg/appsmith/blob/release/app/client/.lintstagedrc) is not getting triggered. If you don't mind can you fix this manually as it is not working automatically? 

Sorry for this issue. We will try to resolve this soon.

**Comment by hetunandu** [comment]
😕
Screenshot 2020-10-27 at 11 07 07 AM

**Comment by JeffResc** [comment]
@hetunandu Huh, I ran `npx eslint --fix`, then `npx prettier --write <file>` and commited my changes. Is that right?

**Comment by hetunandu** [comment]
> @hetunandu Huh, I ran `npx eslint --fix`, then `npx prettier --write <file>` and commited my changes. Is that right?

Yes it should have worked. It always works on our side. Can you add any other details about your issue here #1411

File: app/client/src/pages/common/ServerUnavailable.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import Button from "components/editorComponents/Button";
 import PageUnavailableImage from "assets/images/404-image.png";
+import { APPLICATIONS_URL } from "constants/routes";
 
 const Wrapper = styled.div`
   text-align: center;
   margin-top: 5%;
   .bold-text {
-    font-weight: ${props => props.theme.fontWeights[3]};
+    font-weight: ${(props) => props.theme.fontWeights[3]};
```
Comment (hetunandu) [inline app/client/src/pages/common/ServerUnavailable.tsx:11]:
```suggestion
    font-weight: ${props => props.theme.fontWeights[3]};
```

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**